docs/frontend-system/utility-apis/05-migrating.md

```diff
@@ -0,0 +1,138 @@
+---
+id: migrating
+title: Migrating Utility APIs from the old frontend system
+sidebar_label: Migrating
+# prettier-ignore
+description: Migrating Utility APIs from the old frontend system
+---
+
+If you are migrating your plugins or app over from the old frontend system, there are a few things to keep in mind in regards to utility APIs.
+
+## Overview
+
+- Migrate your repo overall to the latest release of Backstage
+- Follow the plugin migration guide <!-- TODO: Link -->
+- Change your package dependencies from `core-*-api` to `frontend-*-api`
+- Change the imports in your code from `core-*-api` to `frontend-*-api`
+- Keep the TypeScript interface and API ref exported as they were, except possibly reconsidering the choice of ID of the latter
+- Wrap the old API factory call in an extension using `createApiExtension`
+- Make sure that this extension is referenced by your migrated plugin
+
+## Prerequisites
+
+This guide assumes that you first [upgrade your repo](../../getting-started/keeping-backstage-updated.md) to the latest release of Backstage. This ensures that you do not have to fight several types of incompatibilities and updates at the same time.
+
+## Dependency changes
+
+In this article we will discuss some interfaces that you used to import from the `@backstage/core-plugin-api` package. Those are now generally moved over to `@backstage/frontend-plugin-api`, so you will want to update both your `package.json` and your source code. This applies both in your `-react` package and your main plugin package.
+
+First `package.json`. The following commands are examples - note that they refer to `plugins/example`, which you'll have to update to the actual folder name that your package to migrate is in.
+
+```bash title="from your repo root"
+yarn --cwd plugins/example remove @backstage/core-plugin-api ;
+yarn --cwd plugins/example add @backstage/frontend-plugin-api
+```
```

Comment:
As an isolated guide for migrating Utility APIs I feel this is perhaps a bit aggressive. Perhaps we can instead mention that it doesn't matter whether you import `createApiRef` and friends from `core-plugin-api` or `frontend-plugin-api`, but you'll need to use `frontend-plugin-api` to create the extension?